### PR TITLE
add child-share data, .py, and .ipynb 

### PR DESCRIPTION
**Pull request by fedderw**

### CODE REVIEW

File: jb/20210103/split_of_child_share_fixed.py [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+summary.to_csv("children_share_funding_summary.csv.gz",compression='gzip')
+
+# drop rows where funding level is 0
+optimal_poverty = summary.sort_values('poverty_gap').drop_duplicates('funding_billions', keep='first')
+optimal_poverty = optimal_poverty.drop(
+    optimal_poverty[optimal_poverty.funding_billions==0].index
+    ) 
+
+optimal_inequality = summary.sort_values('gini').drop_duplicates('funding_billions', keep='first')
+optimal_inequality = optimal_inequality.drop(
```
Comment (MaxGhenis) [inline jb/20210103/split_of_child_share_fixed.py:137]: would this work? could be clearer. also please add a comment about why this is necessary (not obvious to me)
```
optimal_inequality = optimal_inequality[optimal_inequality.funding_billions > 0]
```

File: jb/20210103/split_of_child_share_fixed.py [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+def big_percent_row(row):  
+    return ubi2(row.funding_billions, row.child_percent_ubi)
+
+summary2[['poverty_rate', 'gini', 'poverty_gap', 'percent_better_off', 'adult_ubi', 'child_ubi']] = summary2.apply(big_percent_row, axis=1)
+summary2
+
+# calculate monthly payments
+summary2['monthly_child_ubi'] =summary2['child_ubi'].apply(lambda x: int(round(x/12,0)))
+summary2['monthly_adult_ubi'] =summary2['adult_ubi'].apply(lambda x: int(round(x/12,0)))
+summary2.sample(10)
```
Comment (MaxGhenis) [inline jb/20210103/split_of_child_share_fixed.py:158]: remove this and any other code that doesn't produce the result dataframe(s) used in the notebook (e.g. also graphing code)

File: jb/20210103/split_of_child_share_fixed.py [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+
+### CALCULATIONS ###
+
+child_allowance_overall = []
+child_allowance_child = []
+child_allowance_adults = []
+
+# Determine the poverty rate impact of a Child Allownace from $0 in new spending to $1 trillion.
+
+for spending in range(0, 1000000000001, 50000000000):
```
Comment (MaxGhenis) [inline jb/20210103/split_of_child_share_fixed.py:193]:
```suggestion
for spending in np.arange(0, 1e12 + 1, 5e10):
```
also below

File: jb/20210103/split_of_child_share_fixed.py [hunk 4]
```diff
@@ -0,0 +1,362 @@
...
+  program['ubi_type'] = program.ubi_type.map({'child_allowance': 'Child allowance',
+                                      'adult_ubi': 'Adult UBI',
+                                      'all_ubi': 'All UBI'})
+  return program
+
+program_overall = melt_dict(overall)
+program_child = melt_dict(child)
+program_adult = melt_dict(adult)
+
+def line_graph(df, x, y, color, title, xaxis_title, yaxis_title):
```
Comment (MaxGhenis) [inline jb/20210103/split_of_child_share_fixed.py:323]: remove

File: jb/20210103/split_of_child_share_fixed.py [hunk 5]
```diff
@@ -0,0 +1,362 @@
+# -*- coding: utf-8 -*-
```
Comment (MaxGhenis) [inline jb/20210103/split_of_child_share_fixed.py:1]: remove this stuff, including pip installs

**Review by MaxGhenis — COMMENTED** [review]
The notebook looks great. I left a few things to do at some point, ideally before launching, but some things we need before launching include:
1. Add the post to the `toc.yml` file so that it's part of the blog
2. Rename the notebook to something URL-friendly, e.g. `child-ubi-share.ipynb`
3. Rename the notebook path to the current date (we can't publish today so can hold until we do, but we can go for Tuesday)
4. Capitalize Gini in charts
5. Remove "UBI" from legend in final chart (it's getting cut off)
6. Only capitalize first word in axis titles in two final charts
7. Use Roboto font in entire chart in last two charts, using `update_layout` as in first charts
8. `hide-input` cell tags (I can't tell if this is done without cloning so thanks if it's already been done)

It'd also be great to format the code with `black`, as there are some long lines and other things that make it hard to read.

**Comment by MaxGhenis** [comment]
cc @ngpsu22

**Comment by fedderw** [comment]
@MaxGhenis Forgive me for this newbie question, but why is it that when I save changes I've to a file in VSCode, when I close the file and open try to open it in Jupyter Lab, the changes aren't there? 

Of course then when I go back and open in VSCode, the changes also aren't there. Yet some changes from yesterday did stick, and I didn't run use any git commands. 

I assume shouldn't be pushing changes that I don't want to be reviewed, only once I am done working.

File: jb/20210103/split_of_child_share_fixed.py (hunk 1 above)
Comment (fedderw) [reply jb/20210103/split_of_child_share_fixed.py:137]: @MaxGhenis It makes the graphs look jumpy and weird, I'd include a screenshot if I could. 

@ngpsu22 It looks like we do this in the notebook as well, the code was just after writing the file, so I'll get rid of it in the notebook as it's now redundant

**Comment by MaxGhenis** [comment]
Try renaming files with `git mv oldfile newfile` if they're already tracked in git

File: jb/20210103/split_of_child_share_fixed.py (hunk 4 above)
Comment (fedderw) [reply jb/20210103/split_of_child_share_fixed.py:323]: deleting all of this, old code from july post that Nate was using for reference

File: jb/20210103/split_of_child_share_fixed.py [hunk 6]
```diff
@@ -0,0 +1,362 @@
...
+  gini = mdf.gini(target_persons, 'new_spm_resources_pp', w='person_weight')
+
+  # Percent winners
+  target_persons['better_off'] = (target_persons.new_spm_resources > target_persons.spm_resources)
+  total_better_off = (target_persons.better_off * target_persons.person_weight).sum()
+  percent_better_off = total_better_off / pop
+
+  return pd.Series([poverty_rate, gini, poverty_gap, percent_better_off, adult_ubi, child_ubi])
+
+def ubi2(funding_billions=0, child_percent_ubi=0):
```
[the remark this answers is not in the document]
Comment (fedderw) [reply jb/20210103/split_of_child_share_fixed.py:83]: New form: 
`def ubi(funding_billions=0, child_percent_funding=None, child_percent_ubi=None):
    """
    args:
        funding_billions: total annual funding for UBI program in billions USD
        child_percent_funding: percent (in whole numbers) of annual funds earmarked
        for child beneficiaries
        child_percent_ubi: proportion of child beneficiary's UBI benefit compared to adult
        beneficiary's payment

    returns:
        series of elements:
         poverty_rate: headcount poverty rate using Supplemental Poverty Measure threshold
         gini: gini coefficient of income
         poverty_gap: SPM unit poverty gap
         percent_better_off: percent of population with higher income after new transfers and taxes
         adult_ubi: annual per adult benefit in dollars
         child_ubi: annual per adult benefit in dollars
    """
    funding = funding_billions * 1e9

    if child_percent_funding is not None:
        child_percent_funding /= 100
        adult_ubi = ((1 - child_percent_funding) * funding) / adult_pop
        child_ubi = (child_percent_funding * funding) / child_pop
    else:
        child_percent_ubi /= 100
        adult_ubi = funding / (adult_pop + (child_pop * child_percent_ubi))
        child_ubi = adult_ubi * `child_percent_ubi``

File: jb/20210103/split_of_child_share_fixed.py (hunk 2 above)
Comment (fedderw) [reply jb/20210103/split_of_child_share_fixed.py:158]: keeping the monthly, calculation, got rid of line 158

**Comment by fedderw** [comment]
@MaxGhenis 
> Try renaming files with `git mv oldfile newfile` if they're already tracked in git

I'm getting this error:. I don't know why it thinks I want to do what it thinks I'm asking.
`C:\Users\fedde\UBICenter\blog\jb>git mv 20210103 20210119`
`Rename from 'jb/20210103' to 'jb/20210119/20210103' failed. Should I try again? (y/n)`

**Comment by fedderw** [comment]
> @MaxGhenis
> 
> > Try renaming files with `git mv oldfile newfile` if they're already tracked in git
> 
> I'm getting this error:. I don't know why it thinks I want to do what it thinks I'm asking.
> `C:\Users\fedde\UBICenter\blog\jb>git mv 20210103 20210119`
> `Rename from 'jb/20210103' to 'jb/20210119/20210103' failed. Should I try again? (y/n)`

It looks like I created a folder with the name 20210119 in an earlier attempt, but for some reason it says the file is in use when I try to delete in file explorer, it says the folder is in use but I've closed out of everything. Although I'm not sure that's the right way regardless.

**Comment by MaxGhenis** [comment]
Thanks @fedderw, just made some minor edits along the lines of my comments. This is ready to publish, but given the inauguration I changed the date to Thursday. I'll merge this first thing Thursday morning.

If you have a few minutes for some minor cleanup, here are two optional last suggestions:
* Avoid hard-coding the population numbers (e.g. pull/export them from the preprocessing script)
* Create functions to add traces in cases where the code is repetitive

This hover card magic is really awesome, I think this post will be a hit.

**Pull request merged**